### ISSUE TITLE
A python implementation

### ISSUE THREAD

**Issue by rahulrajpl** [mikeliturbe/pasad#1]
Hello everyone,
I just wanted to contribute a python implementation of PASAD. Making it available for academic purpose only. Here is the link
https://github.com/rahulrajpl/PASAD

Kindly open pull request in case of any error or enhancement.

**Comment by wissamaoudi** [comment]
Hello Rahul,

Thanks for your contribution and effort! A python implementation is always
useful.

It's a pretty neat work, I only have one advice: When plotting the scree
plot, do not display the first eigenvalue because it is
typically disproportionately larger than the others, which makes it
overshadow other eigenvalues that could be interesting. You could get
better detection results with r > 1. Usually the first (leading) 10-20
eigenvalues carry essential information about the signal, and this can
only be visualized if you skip plotting the first eigenvalue. As to how to
decide which r eigenvalues are enough, look for an early "break" like the
possible ones shown in the attached figure.

[image: screeplot_ex.png]

Best,
Wissam

**Comment by rahulrajpl** [comment]
Hello Wissam,

Thank you very much for the input. I could not figure out how to exactly reproduce the graphs shown in the paper and this is the closest I could obtain. If I may ask a doubt, what could be the reason for such disproportionate first eigen value? Is it an inherent characteristic of trajectory matrix of a time series data?

Regards,
Rahul

**Comment by wissamaoudi** [comment]
Yes, based on extensive experiments, it is probably an inherent characteristic of the trajectory matrix. It seems that most of the variance in time series data generated from dynamical systems can be attributed to a single degree of freedom.

Best,
Wissam
